flink-connector-opensearch/src/main/java/org/apache/flink/connector/opensearch/sink/OpensearchSink.java

```diff
@@ -88,4 +88,8 @@ public SinkWriter<IN> createWriter(InitContext context) throws IOException {
                 context.metricGroup(),
                 context.getMailboxExecutor());
     }
+
+    DeliveryGuarantee getDeliveryGuarantee() {
```

Comment:
add `@VisibleForTesting`

Reply:
The architecture tests verify that these methods are only called in tests ;) It allows us to leak some implementation details into tests without impacting production code. Can make your life a lot easier.

Reply:
I believe in APIs :) I think there are number of ways to approach testing of implementation details, but that's certainly very opinionated subjects, thanks for sharing your point of view.